spec/ruby/optional/capi/encoding_spec.rb

```diff
@@ -721,4 +721,25 @@
       str.bytes.should == [0, 0x24]
     end
   end
+
+  describe "rb_define_dummy_encoding" do
+    it "defines the dummy encoding" do
+      @s.rb_define_dummy_encoding("FOO")
+      enc = Encoding.find("FOO")
+      enc.should.dummy?
+    end
+
+    it "returns the index of the dummy encoding" do
+      index = @s.rb_define_dummy_encoding("BAR")
+      index.should == Encoding.list.size - 1
+    end
+
+    ruby_version_is "3.2" do
+      it "raises EncodingError if too many encodings" do
+        -> {
+          1_000.times {|i| @s.rb_define_dummy_encoding("R_#{i}") }
+        }.should raise_error(EncodingError, "too many encoding (> 256)")
```

Comment:
This should be done in a subprocesss with `ruby_exe` because it's global state that cannot be undone.
For instance it could cause Encoding#replicate specs (if they are run after) to fail.